src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -1280,6 +1297,8 @@ private static ImmutableArray<StateForCase> RemoveEvaluation(ImmutableArray<Stat
                                 out trueTestPermitsTrueOther, out falseTestPermitsTrueOther, out trueTestImpliesTrueOther, out falseTestImpliesTrueOther);
                             break;
                         case BoundDagValueTest v2:
+                            if (!v2.IsSubsumptionOnly)
+                                foundExplicitValueTest = true;
```

Comment:
I think to make sure the test is going to be actually merged, we should add:

`if (trueTestImpliesTrueOther || falseTestImpliesTrueOther)`

Since there's no failing test without this change, it's seems like either there's a test gap or it gets handled in some other way. Without tracing, I can't say what happens if that's false.